packages/api/src/routes/medical/patient.ts

```diff
@@ -430,4 +438,61 @@ router.get(
   })
 );
 
+/** ---------------------------------------------------------------------------
+ * PUT /patient/:id/opting-out
+ *
+ * Returns whether the patient is opted out of data pulling and sharing.
+ *
+ * @param req.cxId The customer ID.
+ * @param req.param.patientId The ID of the patient whose data is to be returned.
+ * @param req.query.optingOut Boolean value to opt patient out or in.
+ */
+router.put(
+  "/opting-out",
+  requestLogger,
+  asyncHandler(async (req: Request, res: Response) => {
+    const { cxId, patient } = getPatientInfoOrFail(req);
+    const optingOut = isTrue(getFrom("query").orFail("optingOut", req));
+
+    const result = await optingPatientInOrOut({ cxId, patientId: patient.id, optingOut });
```

Comment:
Let's name the command with the action being done, please. Something like `setHieOptOut` or similar.